backend/src/main/java/com/zpi/backend/reservations/ReservationService.java

```diff
@@ -243,4 +284,61 @@ public List<String> getReservationStatuses(Authentication authentication, String
         }
         return null;
     }
+
+    // Scheduling tasks
+
+    @Scheduled(cron = "0 0 8 ? * *", zone = "Europe/Warsaw")
+    public void sendRemaindersAboutReservation2DaysBefore() throws IOException {
+        logger.info("Sending reservation reminders 2 days before.");
+        List<Reservation> reservationsSoon = reservationRepository.getReservationsStartingInTwoDays();
+        for (Reservation r: reservationsSoon){
+            Context context = emailService.getRemindingIn2DaysEmailContext(
+                    r.getReservationId(), r.getGameInstance().getGame().getName()
+            );
+            emailService.sendEmailWithHtmlTemplate(
+                    r.getRenter(),
+                    context.getVariable("pl_title").toString(),
+                    EmailService.EMAIL_TEMPLATE,
+                    context,
+                    EmailType.RESERVATION_COMING_SOON);
+        }
+    }
+
+    @Scheduled(cron = "0 0 8 ? * *", zone = "Europe/Warsaw")
+    public void sendRemaindersAboutReservationOnDay() throws IOException {
+        logger.info("Sending reservation reminders on reservation day.");
+        List<Reservation> reservationsToday = reservationRepository.getReservationsStartingToday();
+        for (Reservation r: reservationsToday){
+            Context context = emailService.getRemindingTodayEmailContext(
+                    r.getReservationId(), r.getGameInstance().getGame().getName()
+            );
+            emailService.sendEmailWithHtmlTemplate(
+                    r.getRenter(),
+                    context.getVariable("pl_title").toString(),
+                    EmailService.EMAIL_TEMPLATE,
+                    context,
+                    EmailType.RESERVATION_TODAY);
+        }
+    }
+
```

Comment:
i've heard thats it's not a good practice to run all cron jobs at the same time, maybe change so that one part of emails are sent on 8 and the other on 8:30 or 8:15

Reply:
ok, I'll change that